apps/api/src/app/inbox/inbox.controller.ts

```diff
@@ -67,32 +80,41 @@ export class InboxController {
     private getInboxPreferencesUsecase: GetInboxPreferences,
     private updatePreferencesUsecase: UpdatePreferences,
     private snoozeNotificationUsecase: SnoozeNotification,
-    private unsnoozeNotificationUsecase: UnsnoozeNotification
+    private unsnoozeNotificationUsecase: UnsnoozeNotification,
+    private parseEventRequest: ParseEventRequest
   ) {}
 
+  @KeylessAccessible()
   @Post('/session')
   async sessionInitialize(
     @Body() body: SubscriberSessionRequestDto,
     @Headers('origin') origin: string
   ): Promise<SubscriberSessionResponseDto> {
-    // TODO: Backward compatibility support - remove in future versions (see NV-5801)
-    const subscriber: SubscriberDto | {} =
-      typeof body.subscriber === 'string' ? { subscriberId: body.subscriber } : body.subscriber || {};
-    const subscriberId: string | undefined = body.subscriberId || (subscriber as SubscriberDto).subscriberId;
+    const subscriber = this.buildSubscriber(body);
 
     return await this.initializeSessionUsecase.execute(
       SessionCommand.create({
-        subscriber: {
-          ...subscriber,
-          subscriberId,
-        } satisfies SubscriberDto,
+        subscriber,
         applicationIdentifier: body.applicationIdentifier,
         subscriberHash: body.subscriberHash,
         origin,
       })
     );
   }
 
+  private buildSubscriber(body: SubscriberSessionRequestDto): SubscriberDto {
+    if (!body.applicationIdentifier || body.applicationIdentifier.startsWith('pk_keyless_')) {
+      return { subscriberId: 'keyless-subscriber-id' };
```

Comment:
I wonder, why not let the Inbox generate some random one and persist it? So we don't need to modify this API part of it.

Reply:
good question. I was not sure who should be responsible for the subscriber generation. to me, it felt like we would want as little logic in the Inbox as possible, so I moved this logic to the API.
let me know what you think about it.